Challenges/StacksQueues/AnimalShelter.cs

```diff
@@ -0,0 +1,89 @@
+﻿
+using DataStructures.StacksQueues;
+using System.Collections;
+
+
+namespace Challenges.StacksQueues
+{
+    public class AnimalShelter<T> where T : Animal
```

Comment:
This is neat, since it allows us to make a Shelter for a specific kind of Animal, e.g. a `AnimalShelter<Dog>`.